core/src/write.jl

```diff
@@ -328,14 +329,31 @@ end
 function write_arrow(
     path::AbstractString,
     table::NamedTuple,
-    compress::Union{ZstdCompressor, Nothing},
+    compress::Union{ZstdCompressor, Nothing};
+    init::Bool,
 )::Nothing
+    # At the start of the simulation, write an empty table to ensure we have permissions
+    # and fail early.
+    # At the end of the simulation, write all non-empty tables, and remove existing empty ones.
+    if !init && isempty(table.time)
+        try
+            rm(path; force = true)
+        catch
+            @warn "Failed to remove outdated results, file may be locked." path
+        end
+        return nothing
+    end
     # ensure DateTime is encoded in a compatible manner
     # https://github.com/apache/arrow-julia/issues/303
     table = merge(table, (; time = convert.(Arrow.DATETIME, table.time)))
     metadata = ["ribasim_version" => string(pkgversion(Ribasim))]
     mkpath(dirname(path))
-    Arrow.write(path, table; compress, metadata)
+    try
+        Arrow.write(path, table; compress, metadata)
+    catch
+        @error "Failed to write results, it may be locked." path
```

Comment:
nit: Re-use string from l342? And maybe (lazy) interpolate the string and include the (now not-catched) error message? Like: `"Failed to write results to $(path): $e"`?

Reply:
That is a slightly different error, so I prefer a different message. Other comments are all fixed in db1ed9201e9221980caa1d7bb71f0dc02c8be466